.github/workflows/rust-unused-dependencies.yml

```diff
@@ -0,0 +1,46 @@
+name: Rust - Unused dependencies
+on:
+    pull_request:
+        paths:
+            - .github/workflows/rust-unused-dependencies.yml
+            - '**/*.rs'
+            - '**/Cargo.toml'
+    # Check if requested manually from the Actions tab
+    workflow_dispatch:
+jobs:
+    cargo-udeps:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Checkout binaries submodule
+              run: git submodule update --init --depth=1 dist-assets/binaries
+
+            - name: Install build dependencies
+              run: |
+                sudo apt-get update
+                sudo apt-get install libdbus-1-dev
+
+            # cargo-udeps can compile on stable, but not run on stable.
+            # So we need a nighly toolchain. But `rust-cache@v1` keys
+            # the cache on the rustc version. So if we just use `nightly`
+            # the cache will be lost every day. To mitigate this we lock
+            # it to a specific nightly. Obviously we have to bump this from
+            # time to time or when `cargo-udeps` release a version incompatible
+            # with our nightly version.
```

Comment:
Another solution to this hack is to install both stable and nightly rustc and install `cargo-udeps` with stable (and have it cached under stable). That way the cache is only invalidated every six weeks (new Rust release). And then we run it under nightly. Installing toolchains is fast, so this is no problem. But if we stay on rolling `nightly` we risk having sudden breakage, which is also not desirable.